test/test_transforms_v2_refactored.py

```diff
@@ -1634,3 +1637,144 @@ def test_transform_negative_degrees_error(self):
     def test_transform_unknown_fill_error(self):
         with pytest.raises(TypeError, match="Got inappropriate fill arg"):
             transforms.RandomAffine(degrees=0, fill="fill")
+
+
+class TestCutMixMixUp:
+    # TODO: Does it work when labels are already dirichlet-distributed? like does Compose([Mixup(), CutMix()]) work?
+    @pytest.mark.parametrize("T", [transforms.Cutmix, transforms.Mixup, "Compose"])
+    @pytest.mark.parametrize("one_hot", [True, False])
+    def test_supported_input_structure(self, T, one_hot):
+
+        num_categories = 10
+        batch_size = 32
+        H, W = 12, 12
+
+        preproc = transforms.Compose([transforms.PILToTensor(), transforms.ToDtype(torch.float32)])
+        if one_hot:
+
+            class ToOneHot(torch.nn.Module):
+                def forward(self, inpt):
+                    img, label = inpt
+                    return img, torch.nn.functional.one_hot(label, num_classes=num_categories)
+
+            preproc = transforms.Compose([preproc, ToOneHot()])
+
+        dataset = FakeData(size=batch_size, image_size=(3, H, W), num_classes=num_categories, transforms=preproc)
+        if T == "Compose":
+            cutmix = transforms.Cutmix(alpha=0.5, num_categories=num_categories)
+            mixup = transforms.Mixup(alpha=0.5, num_categories=num_categories)
+            cutmix_mixup = transforms.Compose([cutmix, mixup])
+            expected_num_non_zero_labels = 3  # see common_checks
+        else:
+            cutmix_mixup = T(alpha=0.5, num_categories=num_categories)
+            expected_num_non_zero_labels = 2  # see common_checks
+
+        dl = DataLoader(dataset, batch_size=batch_size)
+
+        # Input sanity checks
+        img, target = next(iter(dl))
+        assert isinstance(img, torch.Tensor) and isinstance(target, torch.Tensor)
+        assert target.shape == (batch_size, num_categories) if one_hot else (batch_size,)
+
+        def check_output(img, target):
+            print(target)
+            assert img.shape == (batch_size, 3, H, W)
+            assert target.shape == (batch_size, num_categories)
+            torch.testing.assert_close(target.sum(axis=-1), torch.ones(batch_size))
+            # Below we check the number of non-zero values in the target tensor.
+            # When just CutMix() (or just MixUp()) is called, we should expect 2
+            # non-zero label values per sample. Although, it may happen that
+            # only 1 non-zero value is present, basically if the transform had
+            # no effect. Here we make sure that:
```

Comment:
Sooo I'm writing this because this is what I observed. But in practice I' not sure when this happens and whether this is expected. Is it because the `Beta` distribution might end up on a value of `1` for `lam`?

CC @pmeier

Reply:
I mean, it is possible, but vanishingly unlikely. For the default value of `alpha=0.5`, the values concentrate around 0 and 1, but hitting a value that is close enough to vanish inside the floating point precision is rather unlikely.

If we want to make sure, we can always fix the random seed or mock the distribution to return `0.5` or the like.

Lastly, we could also set `alpha=10` or the like, which heavily concentrates distribution around `0.5` and thus again making the no-op case highly unlikely.